packages/tocco-util/src/actions/modules/preActions/preAction.js

```diff
@@ -0,0 +1,21 @@
+import {call} from 'redux-saga/effects'
+
+export default preAction =>
+  function* (definition, ids) {
+    let abort = false
+    let i = 0
+    let params = {}
+
+    while (!abort && preAction[i]) {
+      const handler = preAction[i]
+      if (handler.shouldRun(definition)) {
+        const response = yield call(handler.run, params, definition, ids)
+
+        abort = response.abort
+        params = {...(params || {}), ...response.params}
```

Comment:
`params = {...params, ...respones.params}` should work. `params` is initialized on line 7